components/event-banner/EventBanner.vue

```diff
@@ -1,28 +1,37 @@
 <template functional>
   <div class="event-banner">
-    <div class="event-banner__banner">
-      <img
-        v-if="props.bannerIcon"
-        :src="props.bannerIcon.url"
-        :width="props.bannerIcon.width"
-        :height="props.bannerIcon.height"
-        alt=""
-        class="event-banner__icon"
-      >
-      <h2
-        class="event-banner__title"
-        :class="{
-          'sr-only': props.visuallyHideName
-        }"
-      >
-        <span v-html="props.name" />
-      </h2>
-      <p class="event-banner__date">
-        {{ props.displayDate }}
-      </p>
+    <div class="animator__clip animator__clip-container">
+      <div class="animator__clip-content event-banner__banner">
+        <span class="animator__fade event-banner__gradient" />
+        <span class="event-banner__wave" />
+        <img
```

Comment:
Use a directive here? https://vuejs.org/v2/guide/custom-directive.html

```vue
<img
  [all existing props]
  v-animator:[stagger]="slide-in"
/>
```

The directive adds the class and `animation-stagger` attribute